packages/babel-helpers/src/helpers/applyDecs.js

```diff
@@ -446,13 +446,19 @@ function applyMemberDecs(
       base = Class;
       metadataMap = staticMetadataMap;
       kind = kind - 5 /* STATIC */;
-
-      initializers = staticInitializers;
+      // initialize staticInitializers when we see a non-field static member
+      if (kind !== 0 /* FIELD */) {
+        staticInitializers = staticInitializers || [];
+        initializers = staticInitializers;
```

Comment:
Nit, but this probably produces a smaller output: `initializers = staticInitializers = staticInitializers || []`.

Reply:
Ah no it's ok, the minifier step already produces this code.

Reply:
> initializers = staticInitializers = staticInitializers || []

This is the exact output in `helpers-generated` by terser, so I don't bother to manually minify here.